src/dist/dist.rs

```diff
@@ -219,6 +220,25 @@ impl Deref for TargetTriple {
     }
 }
 
+fn is_32bit_userspace() -> bool {
+    // Check if /bin/sh is a 32-bit binary. If it doesn't exist, fall back to
+    // checking if _we_ are a 32-bit binary.
+    // rustup-init.sh also relies on checking /bin/sh for bitness.
+
+    fn inner() -> io::Result<bool> {
```

Comment:
Maybe consider adding a note that the inner function is used to simplify error handling?

Reply:
Done